src/Core/Addon/Module/ModuleRepository.php

```diff
@@ -460,21 +460,22 @@ public function getModuleAttributes($name)
     }
 
     /**
+     * Send request to get module details on the marketplace, then merge the data received in Module instance
      * @param $moduleId
      * @return Module
      */
     public function getModuleById($moduleId)
     {
         $moduleAttributes = $this->adminModuleProvider->getModuleAttributesById($moduleId);
-        $attributes = $this->getModuleAttributes($moduleAttributes['name']);
+        $module = $this->getModule($moduleAttributes['name']);
 
-        foreach ($attributes->all() as $name => $value) {
-            if (!array_key_exists($name, $moduleAttributes)) {
-                $moduleAttributes[$name] = $value;
+        foreach ($moduleAttributes as $name => $value) {
+            if (!$module->attributes->has($name)) {
+                $module->attributes->set($name, $value);
             }
         }
 
-        return new Module($moduleAttributes);
```

Comment:
Recreating the module instance without asking for the ModuleRepository make the data about the module installation empty. That's why the "Install" button was always displayed.